packages/dd-trace/src/encode/tags-processors.js

```diff
@@ -0,0 +1,121 @@
+// From agent truncators: https://github.com/DataDog/datadog-agent/blob/main/pkg/trace/agent/truncator.go
```

Comment:
I don't understand the goal of this file. Pretty much everything it does should already have been handled before it ever reaches this code path.

Reply:
is everything done already? 🤔 Where are keys and values in `meta` and `metrics` limited in size? Do we also have defaults for service name and span name?

Reply:
I'm not 100% sure everything is handled, but basically the idea is that everything _should_ be handled already, and if not they should be. I think in some cases some of these limits were put in integrations directly instead of in the tracer, so it might be worth it to move everything to the tracer directly.

Reply:
you mean move this to the `format`, before it reaches the encoder?